### PR TITLE
Add GitHub action for windows guest integration tests

### PR DESCRIPTION
**Pull request by rveerama1**

### CODE REVIEW

File: .github/workflows/integration-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
+          echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ jammy main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
+          sudo apt update
+          sudo apt install -y azure-cli
+      - name: Download assets
+        run: |
+          mkdir $HOME/workloads
+          az storage blob download --container-name private-images --file "$HOME/workloads/windows-server-2022-amd64-2.raw" --name windows-server-2022-amd64-2.raw --connection-string "${{ secrets.CH_PRIVATE_IMAGES }}"
+      - name: Run Windows guest integration tests
+        timeout-minutes: 60
```
Comment (likebreath) [inline .github/workflows/integration-windows.yaml:35]: They normally take 3-5 mins, so a timeout of 10 mins should be enough.

File: .github/workflows/integration-windows.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        run: scripts/dev_cli.sh tests --integration-windows
+      - name: Run Windows guest integration tests for musl
+        timeout-minutes: 60
```
Comment (likebreath) [inline .github/workflows/integration-windows.yaml:38]: Ditto.

File: .github/workflows/integration-windows.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: Cloud Hypervisor Tests (Wingows Guest)
+on: [pull_request, create]
+
+jobs:
+  build:
+    if: github.event_name == 'pull_request'
+    name: Tests (Windows Guest)
+    runs-on: garm-jammy
```
Comment (likebreath) [inline .github/workflows/integration-windows.yaml:8]: Just a reminder that Windows guest tests are also tested on aarch64, with musl build only though.

File: .github/workflows/integration-windows.yaml (hunk 1 above)
Comment (rveerama1) [reply .github/workflows/integration-windows.yaml:35]: Picked those values from Jenkinsfile. I will change it once build goes through.

**Comment by rveerama1** [comment]
@rbradford  This PR can not get the secret which you added at Organization level.

I have added CH_PRIVATE_IMAGES on my account at repository level. My trails were able to get the secret.

File: .github/workflows/integration-windows.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+name: Cloud Hypervisor Tests (Wingows Guest)
+on: [pull_request, create]
+
+jobs:
+  build:
+    if: github.event_name == 'pull_request'
```
Comment (rbradford) [inline .github/workflows/integration-windows.yaml:6]: Don't need this.

File: .github/workflows/integration-windows.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+name: Cloud Hypervisor Tests (Wingows Guest)
```
Comment (rbradford) [inline .github/workflows/integration-windows.yaml:1]: Wingows - it's going, where does it go? ...! :-)

**Comment by rbradford** [comment]
> @rbradford This PR can not get the secret which you added at Organization level.
> 
> I have added CH_PRIVATE_IMAGES on my account at repository level. My trails were able to get the secret.

This intentional GitHub behaviour:

https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

I'll take your PR and move it to a branch and create a PR from that.

**Pull request closed without merging**